src/Lecture2.hs

```diff
@@ -260,8 +321,22 @@ data EvalError
 {- | Having all this set up, we can finally implement an evaluation function.
 It returns either a successful evaluation result or an error.
 -}
+
+getResultInt :: Maybe Int -> Int
+getResultInt (Just n) = n
+
+sumEither :: Either EvalError Int -> Either EvalError Int -> Either EvalError Int
+sumEither (Right x) (Right y) = Right (x+y)
+sumEither (Left (VariableNotFound str)) _ = Left (VariableNotFound str)
+sumEither _ (Left (VariableNotFound str)) = Left (VariableNotFound str)
```

Comment:
Once you learn about monads, you'll be able to write this function even shorter! 🙂